packages/devextreme/js/ui/chat.d.ts

```diff
@@ -151,6 +160,20 @@ export type Message = {
     typing?: boolean;
 };
 
+/** @public */
+export type MessageTemplateData = {
+    readonly component: dxChat;
+    readonly message?: Message;
+    readonly text?: string;
+};
+
+/**
+ * @docid
+ * @namespace DevExpress.ui.dxChat
+ * @public
+ */
+export type MessageTemplate = ((data: MessageTemplateData, messageBubbleElement: DxElement) => string | UserDefinedElement);
```

Comment:
why do you write `messageBubbleElement`?
I thought it's just `container` in all out templates

Reply:
no, in .d.ts there are different names, but not container, like `itemElement`, `groupElement`,`fieldElement`, `contentElement` for different templates. I decided it's ok to name it `messageBubbleElement` in this case

p.s. just to make it clear, it's not related to a name we use in templates code, it's just a second arg of function and user can name it whatever he wants

Reply:
Do you really need a new standalone (docid - documentable) type for a template? AFAIK it is a single case for a such extraction

Reply:
Changed to inline